### ISSUE TITLE
disk usage increased by 1 without a reservation

### ISSUE THREAD

**Issue by RAOF** [koverstreet/bcachefs#122]
This is with bcachefs master e1f6739c rebased on top of 5.6.14 (+ Ubuntu sauce)

There was no particular trigger that I can identify.

```
[ 1451.743482] ------------[ cut here ]------------
[ 1451.743484] disk usage increased by 1 without a reservation
[ 1451.743555] WARNING: CPU: 6 PID: 332 at fs/bcachefs/buckets.c:407 bch2_fs_usage_apply+0xdf/0xf0 [bcachefs]
[ 1451.743555] Modules linked in: thunderbolt rfcomm ccm veth xt_comment dummy xt_CHECKSUM xt_MASQUERADE xt_conntrack ipt_REJECT nf_reject_ipv4 xt_tcpudp ip6table_mangle ip6table_nat iptable_mangle iptable_nat nf_nat nf_conntrack nf_defrag_ipv6 nf_defrag_ipv4 nf_tables nfnetlink bpfilter cmac algif_hash algif_skcipher af_alg bnep binfmt_misc zfs(PO) zunicode(PO) zavl(PO) icp(PO) zcommon(PO) znvpair(PO) spl(O) zlua(PO) mei_hdcp intel_rapl_msr snd_hda_codec_hdmi snd_hda_codec_realtek snd_hda_codec_generic dell_laptop x86_pkg_temp_thermal nls_iso8859_1 ledtrig_audio intel_powerclamp coretemp intel_cstate intel_rapl_perf snd_hda_intel dell_wmi snd_intel_dspcfg serio_raw snd_hda_codec dell_smbios dcdbas intel_wmi_thunderbolt snd_hda_core dell_wmi_descriptor wmi_bmof snd_hwdep snd_pcm ath10k_pci ath10k_core snd_seq_midi snd_seq_midi_event snd_rawmidi ath snd_seq mac80211 snd_seq_device snd_timer snd soundcore cfg80211 btusb mei_me btrtl btbcm libarc4 btintel mei hid_multitouch bluetooth joydev
[ 1451.743572]  input_leds cdc_acm ecdh_generic ecc uvcvideo hid_sensor_magn_3d videobuf2_vmalloc videobuf2_memops hid_sensor_incl_3d videobuf2_v4l2 hid_sensor_gyro_3d hid_sensor_accel_3d hid_sensor_rotation hid_sensor_trigger industrialio_triggered_buffer kfifo_buf hid_sensor_iio_common videobuf2_common industrialio intel_pch_thermal videodev ucsi_acpi processor_thermal_device typec_ucsi mc intel_rapl_common cros_ec_ishtp cros_ec intel_soc_dts_iosf typec intel_vbtn soc_button_array int3403_thermal int340x_thermal_zone mac_hid int3400_thermal intel_hid acpi_pad acpi_thermal_rel sparse_keymap sch_fq_codel kvm_intel kvm iptable_filter ip6table_filter ip6_tables br_netfilter bridge stp llc arp_tables efi_pstore parport_pc ppdev lp parport ip_tables x_tables autofs4 poly1305_generic libpoly1305 poly1305_x86_64 chacha_generic chacha_x86_64 libchacha bcachefs crc64 btrfs blake2b_generic zstd_compress raid10 raid456 async_raid6_recov async_memcpy async_pq async_xor async_tx xor raid6_pq libcrc32c
[ 1451.743588]  raid1 raid0 multipath linear hid_sensor_custom hid_sensor_hub intel_ishtp_loader intel_ishtp_hid wacom usbhid hid_generic rtsx_pci_sdmmc amdgpu i915 crct10dif_pclmul crc32_pclmul ghash_clmulni_intel amd_iommu_v2 gpu_sched ttm i2c_algo_bit drm_kms_helper aesni_intel syscopyarea sysfillrect sysimgblt fb_sys_fops cec crypto_simd rc_core cryptd glue_helper drm nvme i2c_i801 nvme_core rtsx_pci intel_lpss_pci intel_ish_ipc intel_lpss idma64 intel_ishtp i2c_hid virt_dma hid wmi video
[ 1451.743602] CPU: 6 PID: 332 Comm: kworker/6:2 Tainted: P           O      5.6.14+bcachefs.git20200524.e1f6739c-1-generic #1-Ubuntu
[ 1451.743602] Hardware name: Dell Inc. XPS 15 9575/0C32VW, BIOS 1.7.1 07/07/2019
[ 1451.743615] Workqueue: bcachefs bch2_write_index [bcachefs]
[ 1451.743623] RIP: 0010:bch2_fs_usage_apply+0xdf/0xf0 [bcachefs]
[ 1451.743624] Code: b8 ff ff ff ff 4c 89 f0 e9 76 ff ff ff 4c 89 ee 48 c7 c7 38 e6 6d c0 89 4d d4 48 89 55 d8 c6 05 c9 2c 07 00 01 e8 96 40 a2 c2 <0f> 0b 48 8b 55 d8 8b 4d d4 eb c1 66 0f 1f 44 00 00 0f 1f 44 00 00
[ 1451.743625] RSP: 0018:ffffa76dc070f588 EFLAGS: 00010282
[ 1451.743626] RAX: 0000000000000000 RBX: ffff90cc13e2aa00 RCX: 0000000000000000
[ 1451.743626] RDX: 000000000000002f RSI: ffffffff84b92dcf RDI: 0000000000000246
[ 1451.743627] RBP: ffffa76dc070f5b8 R08: ffffffff84b92da0 R09: 000000000000002f
[ 1451.743627] R10: 0000000000000000 R11: 0000000000000001 R12: ffff90ccbf8d0000
[ 1451.743628] R13: 0000000000000001 R14: 0000000000000008 R15: ffff90ccbf8d0000
[ 1451.743629] FS:  0000000000000000(0000) GS:ffff90ccdf780000(0000) knlGS:0000000000000000
[ 1451.743629] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[ 1451.743630] CR2: 00007f9960c54c00 CR3: 00000003597b4003 CR4: 00000000003606e0
[ 1451.743630] Call Trace:
[ 1451.743641]  bch2_trans_fs_usage_apply+0x4c/0xa0 [bcachefs]
[ 1451.743649]  ? bch2_mark_key_locked+0x3b4/0x720 [bcachefs]
[ 1451.743656]  ? bch2_mark_update+0xbd/0x200 [bcachefs]
[ 1451.743664]  ? bch2_replicas_delta_list_apply+0x63/0x160 [bcachefs]
[ 1451.743672]  bch2_trans_commit_write_locked+0x8e2/0x960 [bcachefs]
[ 1451.743681]  ? btree_iter_pos_changed+0x1bd/0x570 [bcachefs]
[ 1451.743689]  do_bch2_trans_commit+0x26d/0x770 [bcachefs]
[ 1451.743696]  __bch2_trans_commit+0x737/0x870 [bcachefs]
[ 1451.743706]  bch2_extent_update+0x2d8/0x4a0 [bcachefs]
[ 1451.743710]  ? __alloc_pages_nodemask+0x169/0x300
[ 1451.743721]  ? __btree_iter_traverse_all+0x132/0x3b0 [bcachefs]
[ 1451.743725]  ? kmalloc_order_trace+0x24/0xa0
[ 1451.743727]  ? _cond_resched+0x19/0x30
[ 1451.743736]  ? __btree_trans_get_iter+0x6b/0x4a0 [bcachefs]
[ 1451.743745]  bch2_write_index_default+0x1e1/0x350 [bcachefs]
[ 1451.743749]  ? __enqueue_entity+0x96/0xa0
[ 1451.743757]  ? bch2_write_index_default+0x71/0x350 [bcachefs]
[ 1451.743761]  ? ttwu_do_activate+0x5b/0x70
[ 1451.743784]  ? __slab_free+0x217/0x330
[ 1451.743785]  ? common_interrupt+0xa/0xf
[ 1451.743787]  ? autoremove_wake_function+0x12/0x40
[ 1451.743790]  ? mempool_free_slab+0x17/0x20
[ 1451.743791]  ? kmem_cache_free+0x288/0x2b0
[ 1451.743792]  ? mempool_free_slab+0x17/0x20
[ 1451.743793]  ? mempool_free+0x2f/0x80
[ 1451.743806]  ? bch2_writepage_io_free+0x15/0x20 [bcachefs]
[ 1451.743817]  ? bch2_bkey_is_incompressible+0x6b/0x130 [bcachefs]
[ 1451.743828]  __bch2_write_index+0x329/0x3c0 [bcachefs]
[ 1451.743835]  bch2_write_index+0x18/0xf0 [bcachefs]
[ 1451.743837]  process_one_work+0x1e8/0x3b0
[ 1451.743838]  worker_thread+0x4d/0x400
[ 1451.743840]  kthread+0x104/0x140
[ 1451.743840]  ? process_one_work+0x3b0/0x3b0
[ 1451.743841]  ? kthread_park+0x90/0x90
[ 1451.743842]  ret_from_fork+0x35/0x40
[ 1451.743844] ---[ end trace f65f629b3dba14ca ]---
[ 1451.743845] bcachefs (nvme0n1p3): disk usage increased more than 8 sectors reserved
[ 1451.743864] bcachefs: bch2_trans_fs_usage_apply() while inserting
[ 1451.743868] bcachefs: bch2_trans_fs_usage_apply() u64s 8 type extent 67382:1291216 snap 0 len 8 ver 155096560: crc: c_size 8 size 8 offset 0 nonce 0 csum 3 compress 5 ptr: 0:1319009127 gen 104
[ 1451.743869] bcachefs: bch2_trans_fs_usage_apply() overlapping with
[ 1451.743871] bcachefs: bch2_trans_fs_usage_apply() u64s 8 type extent 67382:1291224 snap 0 len 128 ver 155089099: crc: c_size 1 size 128 offset 0 nonce 0 csum 3 compress 3 ptr: 0:1318926274 gen 104
[ 1451.743872] bcachefs: bch2_trans_fs_usage_apply() while inserting
[ 1451.743875] bcachefs: bch2_trans_fs_usage_apply() u64s 7 type alloc 0:1288013 snap 0 len 0 ver 0: gen 104 read_time 7527 write_time 5999 data_type 4 dirty_sectors 392 oldest_gen 103
[ 1451.743875] bcachefs: bch2_trans_fs_usage_apply() overlapping with
[ 1451.743877] bcachefs: bch2_trans_fs_usage_apply() u64s 7 type alloc 0:1288013 snap 0 len 0 ver 0: gen 104 read_time 7527 write_time 5999 data_type 4 dirty_sectors 391 oldest_gen 103
[ 1451.743878] bcachefs: bch2_trans_fs_usage_apply() while inserting
[ 1451.743879] bcachefs: bch2_trans_fs_usage_apply() u64s 7 type alloc 0:1288094 snap 0 len 0 ver 0: gen 104 read_time 7527 write_time 5999 data_type 4 dirty_sectors 482 oldest_gen 103
[ 1451.743880] bcachefs: bch2_trans_fs_usage_apply() overlapping with
[ 1451.743881] bcachefs: bch2_trans_fs_usage_apply() u64s 7 type alloc 0:1288094 snap 0 len 0 ver 0: gen 104 read_time 7527 write_time 5999 data_type 4 dirty_sectors 474 oldest_gen 103
```

**Comment by RAOF** [comment]
I think that this gets tickled by the zfs, accessing a loopback image.

```
<4>[15290.908288] ------------[ cut here ]------------
<4>[15290.908289] disk usage increased by 1 without a reservation
<4>[15290.908342] WARNING: CPU: 0 PID: 50224 at fs/bcachefs/buckets.c:407 bch2_fs_usage_apply+0xe0/0xf0 [bcachefs]
<4>[15290.908343] Modules linked in: binfmt_misc veth nft_masq nft_chain_nat bridge stp llc zfs(PO) zunicode(PO) zavl(PO) icp(PO) zcommon(PO) znvpair(PO) spl(O) zlua(PO) ebtable_filter ebtables ip6table_raw ip6table_mangle ip6table_nat ip6table_filter ip6_tables iptable_raw iptable_mangle iptable_nat nf_nat nf_conntrack nf_defrag_ipv6 nf_defrag_ipv4 iptable_filter bpfilter nf_tables nfnetlink unix_diag thunderbolt rfcomm ccm snd_hda_codec_hdmi snd_hda_codec_realtek snd_hda_codec_generic cmac algif_hash algif_skcipher af_alg bnep x86_pkg_temp_thermal intel_powerclamp coretemp kvm_intel kvm snd_hda_intel snd_intel_dspcfg mei_hdcp intel_rapl_msr snd_hda_codec snd_hda_core snd_hwdep crct10dif_pclmul ghash_clmulni_intel snd_pcm nls_iso8859_1 snd_seq_midi dell_laptop ledtrig_audio snd_seq_midi_event aesni_intel crypto_simd snd_rawmidi cryptd glue_helper ath10k_pci intel_cstate amdgpu ath10k_core ath mac80211 snd_seq dell_wmi snd_seq_device input_leds dell_smbios snd_timer i915 dcdbas intel_rapl_perf
dmesg-efi-159315558507001:
Oops#1 Part7
<4>[15290.908358]  wmi_bmof serio_raw snd btusb intel_wmi_thunderbolt dell_wmi_descriptor cfg80211 efi_pstore uvcvideo hid_multitouch btrtl btbcm soundcore amd_iommu_v2 videobuf2_vmalloc btintel gpu_sched videobuf2_memops videobuf2_v4l2 mei_me bluetooth videobuf2_common ttm libarc4 hid_sensor_accel_3d hid_sensor_magn_3d mei videodev hid_sensor_rotation hid_sensor_gyro_3d hid_sensor_incl_3d ecdh_generic mc cdc_acm hid_sensor_trigger joydev ecc industrialio_triggered_buffer kfifo_buf drm_kms_helper hid_sensor_iio_common cec industrialio rc_core i2c_algo_bit fb_sys_fops syscopyarea cros_ec_ishtp sysfillrect cros_ec processor_thermal_device intel_rapl_common ucsi_acpi sysimgblt intel_soc_dts_iosf intel_pch_thermal typec_ucsi typec soc_button_array int3403_thermal int340x_thermal_zone intel_vbtn mac_hid acpi_pad int3400_thermal acpi_thermal_rel intel_hid sparse_keymap sch_fq_codel parport_pc ppdev lp parport drm ip_tables x_tables autofs4 poly1305_x86_64 chacha_x86_64 libchacha bcachefs crc64 xor
<4>[15290.908372]  raid6_pq libcrc32c hid_sensor_custom hid_sensor_hub intel_ishtp_loader intel_ishtp_hid wacom usbhid hid_generic rtsx_pci_sdmmc crc32_pclmul nvme nvme_core i2c_i801 intel_lpss_pci rtsx_pci intel_lpss intel_ish_ipc idma64 intel_ishtp virt_dma i2c_hid wmi hid video
dmesg-efi-159315558506001:
Oops#1 Part6
<4>[15290.908380] CPU: 0 PID: 50224 Comm: kworker/0:1 Tainted: P           O      5.7.0+bcachefs.git20200616.467a7a43-2-generic #1-Ubuntu
<4>[15290.908380] Hardware name: Dell Inc. XPS 15 9575/0C32VW, BIOS 1.7.1 07/07/2019
<4>[15290.908392] Workqueue: bcachefs bch2_write_index [bcachefs]
<4>[15290.908400] RIP: 0010:bch2_fs_usage_apply+0xe0/0xf0 [bcachefs]
<4>[15290.908401] Code: b8 ff ff ff ff 4c 89 f0 e9 75 ff ff ff 4c 89 ee 48 c7 c7 e8 07 47 c0 89 4d d4 48 89 55 d8 c6 05 d8 3a 07 00 01 e8 95 6a e9 c1 <0f> 0b 48 8b 55 d8 8b 4d d4 eb c1 0f 1f 44 00 00 0f 1f 44 00 00 55
<4>[15290.908402] RSP: 0018:ffffa05141ef7570 EFLAGS: 00010282
<4>[15290.908403] RAX: 0000000000000000 RBX: ffff8e06ffdcf900 RCX: 0000000000000000
<4>[15290.908404] RDX: 000000000000002f RSI: ffffffff83daed8f RDI: 0000000000000246
<4>[15290.908404] RBP: ffffa05141ef75a0 R08: ffffffff83daed60 R09: 000000000000002f
<4>[15290.908404] R10: 0000000000000000 R11: 0000000000000001 R12: ffff8e071b7c0000
<4>[15290.908405] R13: 0000000000000001 R14: 0000000000000008 R15: ffff8e071b7c0000
<4>[15290.908406] FS:  0000000000000000(0000) GS:ffff8e071f600000(0000) knlGS:0000000000000000
<4>[15290.908406] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
<4>[15290.908407] CR2: 00007f4ad7fe7000 CR3: 00000002dcc0a006 CR4: 00000000003606f0
<4>[15290.908407] Call Trace:
<4>[15290.908430]  bch2_trans_fs_usage_apply+0x4f/0xa0 [bcachefs]
<4>[15290.908449]  ? bch2_mark_key_locked+0x3c5/0x720 [bcachefs]
<4>[15290.908456]  ? bch2_mark_update+0xbd/0x200 [bcachefs]
<4>[15290.908462]  ? bch2_replicas_delta_list_apply+0x63/0x160 [bcachefs]
<4>[15290.908469]  bch2_trans_commit_write_locked+0x95c/0xa60 [bcachefs]
<4>[15290.908475]  ? btree_iter_pos_changed+0x196/0x630 [bcachefs]
dmesg-efi-159315558505001:
Oops#1 Part5
<4>[15290.908481]  ? bch2_trans_update2.isra.0+0x189/0x1d0 [bcachefs]
<4>[15290.908487]  do_bch2_trans_commit+0x1f6/0x800 [bcachefs]
<4>[15290.908492]  __bch2_trans_commit+0x777/0x8b0 [bcachefs]
<4>[15290.908501]  bch2_extent_update+0x2de/0x4a0 [bcachefs]
<4>[15290.908504]  ? __alloc_pages_nodemask+0x161/0x2f0
<4>[15290.908511]  ? __btree_iter_traverse_all+0x186/0x3c0 [bcachefs]
<4>[15290.908513]  ? kmalloc_order_trace+0x24/0xa0
<4>[15290.908514]  ? _cond_resched+0x19/0x30
<4>[15290.908520]  ? __btree_trans_get_iter+0x6b/0x480 [bcachefs]
<4>[15290.908528]  bch2_write_index_default+0x1e1/0x350 [bcachefs]
<4>[15290.908536]  ? bch2_write_index_default+0x71/0x350 [bcachefs]
<4>[15290.908538]  ? __wake_up_common_lock+0x8a/0xc0
<4>[15290.908540]  ? __slab_free+0x217/0x330
<4>[15290.908541]  ? update_load_avg+0x553/0x630
<4>[15290.908542]  ? record_times+0x1b/0x90
<4>[15290.908543]  ? __enqueue_entity+0x96/0xa0
<4>[15290.908545]  ? mempool_free_slab+0x17/0x20
<4>[15290.908546]  ? kmem_cache_free+0x288/0x2b0
<4>[15290.908547]  ? mempool_free_slab+0x17/0x20
<4>[15290.908547]  ? mempool_free+0x2f/0x80
<4>[15290.908555]  ? bch2_writepage_io_free+0x15/0x20 [bcachefs]
<4>[15290.908563]  ? bch2_bkey_is_incompressible+0x6b/0x130 [bcachefs]
<4>[15290.908570]  __bch2_write_index+0x32d/0x3c0 [bcachefs]
<4>[15290.908577]  bch2_write_index+0x18/0xf0 [bcachefs]
<4>[15290.908578]  process_one_work+0x1e8/0x3b0
<4>[15290.908579]  worker_thread+0x4d/0x400
<4>[15290.908580]  kthread+0x104/0x140
<4>[15290.908581]  ? process_one_work+0x3b0/0x3b0
<4>[15290.908582]  ? kthread_park+0x90/0x90
<4>[15290.908583]  ret_from_fork+0x35/0x40
<4>[15290.908584] ---[ end trace 8304818fc03db150 ]---
<3>[15290.908600] bcachefs (nvme0n1p3): disk usage increased more than 8 sectors reserved
dmesg-efi-159315558504001:
Oops#1 Part4
<3>[15290.908601] bcachefs: bch2_trans_fs_usage_apply() while inserting
<3>[15290.908604] bcachefs: bch2_trans_fs_usage_apply() u64s 8 type extent 42985:4434744 snap 0 len 8 ver 8361501: crc: c_size 8 size 8 offset 0 nonce 0 csum 3 compress 5 ptr: 0:606788608 gen 1
<3>[15290.908604] bcachefs: bch2_trans_fs_usage_apply() overlapping with
<3>[15290.908606] bcachefs: bch2_trans_fs_usage_apply() u64s 8 type extent 42985:4434848 snap 0 len 120 ver 8344172: crc: c_size 1 size 128 offset 8 nonce 0 csum 3 compress 3 ptr: 0:606079539 gen 1
<3>[15290.908607] bcachefs: bch2_trans_fs_usage_apply() while inserting
<3>[15290.908609] bcachefs: bch2_trans_fs_usage_apply() u64s 7 type alloc 0:591874 snap 0 len 0 ver 0: gen 1 read_time 24 write_time 413 data_type 4 dirty_sectors 510
<3>[15290.908610] bcachefs: bch2_trans_fs_usage_apply() overlapping with
<1>[15290.908612] BUG: unable to handle page fault for address: 000000000002975a
<1>[15290.908613] #PF: supervisor instruction fetch in kernel mode
<1>[15290.908614] #PF: error_code(0x0010) - not-present page
<6>[15290.908614] PGD 0 P4D 0 
<4>[15290.908616] Oops: 0010 [#1] SMP PTI
<4>[15290.908617] CPU: 0 PID: 50224 Comm: kworker/0:1 Tainted: P        W  O      5.7.0+bcachefs.git20200616.467a7a43-2-generic #1-Ubuntu
<4>[15290.908618] Hardware name: Dell Inc. XPS 15 9575/0C32VW, BIOS 1.7.1 07/07/2019
<4>[15290.908625] Workqueue: bcachefs bch2_write_index [bcachefs]
<4>[15290.908626] RIP: 0010:0x2975a
<4>[15290.908646] Code: Bad RIP value.
<4>[15290.908647] RSP: 0018:ffffa05141ef7598 EFLAGS: 00010206
<4>[15290.908648] RAX: 000000000002975a RBX: ffffa05141ef7b18 RCX: ffff8e07113949c0
<4>[15290.908648] RDX: ffffffffc040ecf8 RSI: ffffffffc040ecf8 RDI: ffffa05141ef75e8
dmesg-efi-159315558503001:
Oops#1 Part3
<4>[15290.908649] RBP: ffffa05141ef75a0 R08: 000000000000080d R09: ffff8e0704738190
<4>[15290.908649] R10: ffffffff83dae968 R11: ffffa05141ef7430 R12: ffff8e070473b218
<4>[15290.908650] R13: ffff8e07113949c0 R14: ffffa05141ef7610 R15: ffff8e071b7c0000
<4>[15290.908651] FS:  0000000000000000(0000) GS:ffff8e071f600000(0000) knlGS:0000000000000000
<4>[15290.908652] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
<4>[15290.908652] CR2: 0000000000029730 CR3: 00000002dcc0a006 CR4: 00000000003606f0
<4>[15290.908653] Call Trace:
<4>[15290.908660]  ? __bkey_unpack_key+0x24/0x50 [bcachefs]
<4>[15290.908670]  bch2_trans_fs_usage_apply.cold+0xf3/0x1fc [bcachefs]
<4>[15290.908690]  ? bch2_disk_reservation_add+0x228/0x240 [bcachefs]
<4>[15290.908697]  ? bch2_replicas_delta_list_apply+0x63/0x160 [bcachefs]
<4>[15290.908704]  bch2_trans_commit_write_locked+0x95c/0xa60 [bcachefs]
<4>[15290.908711]  ? btree_iter_pos_changed+0x196/0x630 [bcachefs]
<4>[15290.908717]  ? bch2_trans_update2.isra.0+0x189/0x1d0 [bcachefs]
<4>[15290.908723]  do_bch2_trans_commit+0x1f6/0x800 [bcachefs]
<4>[15290.908729]  __bch2_trans_commit+0x777/0x8b0 [bcachefs]
<4>[15290.908738]  bch2_extent_update+0x2de/0x4a0 [bcachefs]
<4>[15290.908740]  ? __alloc_pages_nodemask+0x161/0x2f0
<4>[15290.908747]  ? __btree_iter_traverse_all+0x186/0x3c0 [bcachefs]
<4>[15290.908748]  ? kmalloc_order_trace+0x24/0xa0
<4>[15290.908750]  ? _cond_resched+0x19/0x30
<4>[15290.908756]  ? __btree_trans_get_iter+0x6b/0x480 [bcachefs]
<4>[15290.908764]  bch2_write_index_default+0x1e1/0x350 [bcachefs]
<4>[15290.908771]  ? bch2_write_index_default+0x71/0x350 [bcachefs]
<4>[15290.908773]  ? __wake_up_common_lock+0x8a/0xc0
<4>[15290.908775]  ? __slab_free+0x217/0x330
<4>[15290.908776]  ? update_load_avg+0x553/0x630
dmesg-efi-159315558502001:
Oops#1 Part2
<4>[15290.908777]  ? record_times+0x1b/0x90
<4>[15290.908778]  ? __enqueue_entity+0x96/0xa0
<4>[15290.908779]  ? mempool_free_slab+0x17/0x20
<4>[15290.908781]  ? kmem_cache_free+0x288/0x2b0
<4>[15290.908782]  ? mempool_free_slab+0x17/0x20
<4>[15290.908783]  ? mempool_free+0x2f/0x80
<4>[15290.908790]  ? bch2_writepage_io_free+0x15/0x20 [bcachefs]
<4>[15290.908798]  ? bch2_bkey_is_incompressible+0x6b/0x130 [bcachefs]
<4>[15290.908805]  __bch2_write_index+0x32d/0x3c0 [bcachefs]
<4>[15290.908812]  bch2_write_index+0x18/0xf0 [bcachefs]
<4>[15290.908813]  process_one_work+0x1e8/0x3b0
<4>[15290.908814]  worker_thread+0x4d/0x400
<4>[15290.908816]  kthread+0x104/0x140
<4>[15290.908816]  ? process_one_work+0x3b0/0x3b0
<4>[15290.908818]  ? kthread_park+0x90/0x90
<4>[15290.908819]  ret_from_fork+0x35/0x40
<4>[15290.908820] Modules linked in: binfmt_misc veth nft_masq nft_chain_nat bridge stp llc zfs(PO) zunicode(PO) zavl(PO) icp(PO) zcommon(PO) znvpair(PO) spl(O) zlua(PO) ebtable_filter ebtables ip6table_raw ip6table_mangle ip6table_nat ip6table_filter ip6_tables iptable_raw iptable_mangle iptable_nat nf_nat nf_conntrack nf_defrag_ipv6 nf_defrag_ipv4 iptable_filter bpfilter nf_tables nfnetlink unix_diag thunderbolt rfcomm ccm snd_hda_codec_hdmi snd_hda_codec_realtek snd_hda_codec_generic cmac algif_hash algif_skcipher af_alg bnep x86_pkg_temp_thermal intel_powerclamp coretemp kvm_intel kvm snd_hda_intel snd_intel_dspcfg mei_hdcp intel_rapl_msr snd_hda_codec snd_hda_core snd_hwdep crct10dif_pclmul ghash_clmulni_intel snd_pcm nls_iso8859_1 snd_seq_midi dell_laptop ledtrig_audio snd_seq_midi_event aesni_intel crypto_simd snd_rawmidi cryptd glue_helper ath10k_pci intel_cstate amdgpu ath10k_core ath mac80211 snd_seq dell_wmi snd_seq_device input_leds dell_smbios snd_timer i915 dcdbas intel_rapl_perf
dmesg-efi-159315558501001:
Oops#1 Part1
<4>[15290.908830]  wmi_bmof serio_raw snd btusb intel_wmi_thunderbolt dell_wmi_descriptor cfg80211 efi_pstore uvcvideo hid_multitouch btrtl btbcm soundcore amd_iommu_v2 videobuf2_vmalloc btintel gpu_sched videobuf2_memops videobuf2_v4l2 mei_me bluetooth videobuf2_common ttm libarc4 hid_sensor_accel_3d hid_sensor_magn_3d mei videodev hid_sensor_rotation hid_sensor_gyro_3d hid_sensor_incl_3d ecdh_generic mc cdc_acm hid_sensor_trigger joydev ecc industrialio_triggered_buffer kfifo_buf drm_kms_helper hid_sensor_iio_common cec industrialio rc_core i2c_algo_bit fb_sys_fops syscopyarea cros_ec_ishtp sysfillrect cros_ec processor_thermal_device intel_rapl_common ucsi_acpi sysimgblt intel_soc_dts_iosf intel_pch_thermal typec_ucsi typec soc_button_array int3403_thermal int340x_thermal_zone intel_vbtn mac_hid acpi_pad int3400_thermal acpi_thermal_rel intel_hid sparse_keymap sch_fq_codel parport_pc ppdev lp parport drm ip_tables x_tables autofs4 poly1305_x86_64 chacha_x86_64 libchacha bcachefs crc64 xor
<4>[15290.908839]  raid6_pq libcrc32c hid_sensor_custom hid_sensor_hub intel_ishtp_loader intel_ishtp_hid wacom usbhid hid_generic rtsx_pci_sdmmc crc32_pclmul nvme nvme_core i2c_i801 intel_lpss_pci rtsx_pci intel_lpss intel_ish_ipc idma64 intel_ishtp virt_dma i2c_hid wmi hid video
<4>[15290.908844] CR2: 000000000002975a
<4>[15290.908845] ---[ end trace 8304818fc03db151 ]---
```

**Comment by koverstreet** [comment]
Fixed by 278fe397bb
